pkg/blobfuse-proxy/init.sh

```diff
@@ -33,25 +33,50 @@ then
   $HOST_CMD rm -f /etc/packages-microsoft-prod.deb
 fi
 
-if [ ! -f "/host/usr/bin/blobfuse-proxy" ];then
+updateBlobfuseProxy="true"
+if [ -f "/host/usr/bin/blobfuse-proxy" ];then
+  old=$(sha256sum /host/usr/bin/blobfuse-proxy | awk '{print $1}')
+  new=$(sha256sum /blobfuse-proxy/blobfuse-proxy | awk '{print $1}')
+  if [ "$old" = "$new" ];then
+    updateBlobfuseProxy="false"
+    echo "no need to update blobfuse-proxy"
+  else
+    rm -rf /host/usr/bin/blobfuse-proxy
+    rm -rf /host/var/lib/kubelet/plugins/blob.csi.azure.com/blobfuse-proxy.sock
+  fi
+fi
+
+if [ "$updateBlobfuseProxy" = "true" ];then
   echo "copy blobfuse-proxy...."
   cp /blobfuse-proxy/blobfuse-proxy /host/usr/bin/blobfuse-proxy
   chmod 755 /host/usr/bin/blobfuse-proxy
 fi
 
-if [ ! -f "/host/usr/lib/systemd/system/blobfuse-proxy.service" ];then
+updateService="true"
+if [ -f "/host/usr/lib/systemd/system/blobfuse-proxy.service" ];then
+  old=$(sha256sum /host/usr/lib/systemd/system/blobfuse-proxy.service | awk '{print $1}')
+  new=$(sha256sum /blobfuse-proxy/blobfuse-proxy.service | awk '{print $1}')
+  if [ "$old" = "$new" ];then
+    updateService="false"
+    echo "no need to update blobfuse-proxy.service"
+  else
+    rm -rf /host/usr/lib/systemd/system/blobfuse-proxy.service
```

Comment:
since it would be overwritten later, no need to do this?

Reply:
fixed